Specialization/Blood.lua

```diff
@@ -56,64 +49,112 @@ function DeathKnight:Blood()
 	local runes = DeathKnight:Runes(fd.timeShift);
 	local runicPower = UnitPower('player', RunicPower);
 	local targets = MaxDps:SmartAoe();
-
+	local targetHpPercent = MaxDps:TargetPercentHealth() * 100;
+	fd.targetHpPercent = targetHpPercent;
+	
 	fd.targets = targets;
 	fd.runes = runes;
 	fd.runicPower = runicPower;
 
-	if covenantId == NightFae then
-		BL.DeathAndDecay = BL.DeathsDue;
-	end
-
 	DeathKnight:BloodGlowCooldowns();
+	
+	if talents[COMMON.SoulReaper] and targetHpPercent <= 35 and cooldown[COMMON.SoulReaper].ready then
+		return COMMON.SoulReaper;
+	end
 
-	-- raise_dead;
-	if cooldown[BL.RaiseDead].ready then
-		return BL.RaiseDead;
+	if buff[BL.DancingRuneWeaponBuff].up then
+		return DeathKnight:DancingRuneWeaponUp();
+	end
+	if not buff[BL.DancingRuneWeaponBuff].up then
+		return DeathKnight:DancingRuneWeaponNotUp();
 	end
 
-	-- blooddrinker,if=!buff.dancing_rune_weapon.up&(!covenant.night_fae|buff.deaths_due.remains>7);
-	if talents[BL.Blooddrinker] and
-		cooldown[BL.Blooddrinker].ready and
-		runes >= 1 and
-		currentSpell ~= BL.Blooddrinker and
-		(
-			not buff[BL.DancingRuneWeaponBuff].up and
-				(covenantId ~= NightFae or buff[BL.DeathsDueBuff].remains > 7 )
-		)
-	then
-		return BL.Blooddrinker;
+end
+
+function DeathKnight:DancingRuneWeaponUp()
+	local fd = MaxDps.FrameData;
+	local buff = fd.buff;
+	local runicPower = fd.runicPower;
+	local debuff = fd.debuff;
+	local cooldown = fd.cooldown;
+	local runes = fd.runes;
+	local covenantId = fd.covenant.covenantId;
+	
+	local deathAndDecay;
+	
+	if covenantId == NightFae then
```

Comment:
I was on the fence because I thought to just leave it in place until Nov 28th then remove it. I will go ahead and remove it to be consistent with the rest of the module changes.

Reply:
Yeah and we already started receiving complaints "why is it suggesting covenant stuff???" :D